### DISCUSSION TITLE
Synchronise GitHub Repo on Android Devices

### DISCUSSION THREAD

**Discussion by chribonn** [community/community#102054] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Is there an Android App that will allow me to synchronise a github registry on my android devices (as I do on my computers)?

I installed the official GitHub app but is does not seem to support this function

Thank you

**Comment by fury-05** [comment]
There are many 3rd party apps out their.

See if this one can help : Pocket Git is a Git client for Android that provides a graphical interface for managing repositories.
It allows you to clone, commit, push, and pull changes directly from your Android device.

**Comment by chribonn** [comment]
Thank you

**Comment by chribonn** [comment]
Play Store tells me that Pocket Git was made for an older version of
Android.  I am looking at Termux / not graphical but if I get it to sync a
repo it would be sufficient.

**Comment by naveen-98** [comment, marked as answer]
Hello,

To synchronize a GitHub repository on your Android device, you can use third-party Git client apps available on the Google Play Store. One popular option is **Termux**, a terminal emulator for Android that also provides a Linux-like environment. Here are general steps to achieve this using Termux:

1. **Install Termux:**
   - Install Termux from the [Google Play Store](https://play.google.com/store/apps/details?id=com.termux).

2. **Open Termux:**
   - Launch the Termux app on your Android device.

3. **Install Git:**
   - Inside Termux, install Git by running the following command:
     ```bash
     pkg install git
     ```

4. **Clone the Repository:**
   - Navigate to the directory where you want to clone the GitHub repository and run the `git clone` command:
     ```bash
     git clone https://github.com/username/repository.git
     ```
     Replace "https://github.com/username/repository.git" with the actual URL of your GitHub repository.

5. **Navigate and Synchronize:**
   - Enter the cloned repository directory using the `cd` command:
     ```bash
     cd repository
     ```
   - Perform Git operations such as `git pull` to synchronize changes:
     ```bash
     git pull
     ```

These steps will help you set up Git on your Android device using Termux and synchronize a GitHub repository.

**Comment by chribonn** [comment]
Thanks. I did that. The only different thing was the source.  I used
F-Droid.  I initially installed from Google Play but apt update and apt
upgrade would not update. I found a source that claimed that there are
problems with the google version. When I updated from F-Droid the apt
commands worked.

**Reply by yaayimanalien** [reply]
perhaps try out the obsidian git plugin. here's a quick tutorial on how to get it working - https://www.reddit.com/r/ObsidianMD/comments/17odzjb/obsidian_android_syncing_via_github_in_2023/
